### PR TITLE
return connected peers as providers

### PR DESCRIPTION
**Pull request by jbenet**
Also:
- epictest: added test for bitswap wo routing
- fixed three-legged-cat (i think)

### CODE REVIEW

**Comment by jbenet** [comment]
@briantigerchow 

```
PASS: TestThreeLeggedCat1KBInstantaneous (0.01s)
```

File: exchange/bitswap/network/ipfs_impl.go [hunk 1]
```diff
@@ -99,7 +99,19 @@ func (bsnet *impl) SetDelegate(r Receiver) {
...
+
+	// Since routing queries are expensive, give bitswap the peers to which we
+	// have open connections. Note that this may cause issues if bitswap starts
+	// precisely tracking which peers provide certain keys. This optimization
+	// would be misleading. In the long run, this may not be the most
+	// appropriate place for this optimization, but it won't cause any harm in
+	// the short term.
+	connectedPeers := bsnet.host.Network().Peers()
+	out := make(chan peer.ID, len(connectedPeers)) // just enough buffer for these connectedPeers
+	for _, id := range bsnet.host.Network().Peers() {
```
Comment (whyrusleeping) [inline exchange/bitswap/network/ipfs_impl.go]: why not range over `connectedPeers` what if its value changes during the time between calls? Could hang if a peer is added in the meantime

Comment (jbenet) [inline exchange/bitswap/network/ipfs_impl.go]: good catch. addressed

Comment (briantigerchow) [inline exchange/bitswap/network/ipfs_impl.go]: @whyrusleeping nice catch

**Comment by briantigerchow** [comment]
LGTM

**Pull request merged**